controllers/operatorpolicy_controller.go

```diff
@@ -790,16 +854,46 @@ func (r *OperatorPolicyReconciler) mustnothaveOpGroup(
 		earlyConds = append(earlyConds, calculateComplianceCondition(policy))
 	}
 
-	err := r.Delete(ctx, desiredOpGroup)
-	if err != nil {
-		return earlyConds, changed, fmt.Errorf("error deleting the OperatorGroup: %w", err)
+	relatedOpGroups = leftovers // append the deleting and deleted ones to this
+	anyAlreadyDeleting := false
+	deletionErrs := make([]string, 0)
+
+	for _, opGroupToDelete := range toDelete {
```

Comment:
After a discussion, we agreed that we should only allow the removal behavior of `Keep` and `DeleteIfUnused` for operator group.

Reply:
And, if there are multiple operator groups, just report it as noncompliant and don't try to adjust things.